adagrams/game.py

```diff
@@ -1,11 +1,84 @@
-def draw_letters():
-    pass
+import copy 
+import random
 
-def uses_available_letters(word, letter_bank):
-    pass
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 
+def draw_letters():
+    i = 0
+    ten_letters = []
+    no_change_letters = LETTER_POOL.copy()
+    while i in range(10):
+        random_letter = random.choice(list(LETTER_POOL))
+        if no_change_letters[random_letter] > 0:
+            ten_letters.append(random_letter)
+            no_change_letters[random_letter] -= 1
+            i += 1
+    return ten_letters
+     
+def uses_available_letters(word, letter_bank):
+    letter_count = 0
+    letter_copy = letter_bank[:]
+    for letter in word:
+        if letter.upper() in letter_copy:
```

Comment:
Since letter_copy is a list of letters, the time complexity of checking if `letter` is in the `letter_copy` is O(n) because you need to loop through every letter in letter_copy to find it. 

You can optimize this slightly by creating a letter_frequency dictionary instead of letter_copy where the keys are the letters and the values are frequency of the letter.

Then when you check if a letter is a key of letter_frequency, you reduce your time complexity to O(1).

if letter in letter_frequency:

Take care to use `key in dict syntax`. If you do `key in dict.keys()` you'd reproduce O(n) time complexity since under the hood Python creates a list of keys when you call .keys() and the in operator for lists is linear.